### PR TITLE
DENG-72. Consider empty values in conversion to numeric and fix type …

### PR DESCRIPTION
**Pull request by lucia-vargas-a**
This PR attempts to:
- Correct a type incompatibility in query/schema
- Handle empty values when doing a cast to integer

in  unified_metrics_v1.

Checklist for reviewer:

- [x] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [x] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/telemetry_derived/unified_metrics_v1/query.sql [hunk 1]
```diff
@@ -150,9 +150,36 @@ mobile_with_searches AS (
     unioned.first_seen_date,
     unioned.normalized_os,
     unioned.normalized_os_version,
-    CAST(SPLIT(normalized_os_version, '.')[SAFE_OFFSET(0)] AS NUMERIC) AS os_version_major,
-    CAST(SPLIT(normalized_os_version, '.')[SAFE_OFFSET(1)] AS NUMERIC) AS os_version_minor,
-    CAST(SPLIT(normalized_os_version, '.')[SAFE_OFFSET(2)] AS NUMERIC) AS os_version_patch,
+    CASE
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/unified_metrics_v1/query.sql:153]: There's a cleaner way to do this:

```sql
NULLIF(unioned.normalized_os_version, 0)
```

**Review by kik-kik — APPROVED** [review]
Change looks fine, but in general I'd suggest using `NULLIF` function instead of `CASE` as it's a bit more compact and easier to read.

File: sql/moz-fx-data-shared-prod/telemetry_derived/unified_metrics_v1/query.sql (hunk 1 above)
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/telemetry_derived/unified_metrics_v1/query.sql:153]: Apologies, this was meant to be ```COALESCE(NULLIF(field_name, ""), 0)```

Comment (lucia-vargas-a) [reply sql/moz-fx-data-shared-prod/telemetry_derived/unified_metrics_v1/query.sql]: Awesome, added some changes here!

**Review by kik-kik — APPROVED** [review]
LGTM 👍

**Pull request merged**